examples/controls_example/public/react_controls/data_controls/data_control_editor.tsx

```diff
@@ -210,13 +204,15 @@ export const DataControlEditor = <State extends DataControlEditorState = DataCon
         data-test-subj="control-editor-custom-settings"
       >
         <CustomSettings
-          currentState={editorState}
+          initialState={initialState}
+          field={fieldRegistry[editorState.fieldName].field}
           updateState={(newState) => setEditorState({ ...editorState, ...newState })}
-          setControlEditorValid={setControlEditorValid}
+          setControlEditorValid={setControlOptionsValid}
+          parentApi={controlGroup}
```

Comment:
Sorry I keep changing this 😟 Turns out, the old way (sending in `editorState`) didn't work for options list settings because the options list **needs** to sometimes edit its own state depending on the values of other state - so, this was causing an infinite `useEffect` loop 🫠 This SHOULD be the last time I touch the props to the `CustomSettings` component